tests/test_georaster.py

```diff
@@ -1814,6 +1854,116 @@ def test_polygonize(self) -> None:
+            # so this particular case is treated differently in tests
+            if target_values is not None and target_values[0] == 112 and raster.filename is not None:
+
+                # Get index and number of not almost equal point (tolerance of 10-4)
+                ind_not_almost_equal = np.abs(gdal_proximity - geoutils_proximity) > 1e-04
+                nb_not_almost_equal = np.count_nonzero(ind_not_almost_equal)
+                # Check that this is a minority of points (less than 0.5%)
+                assert nb_not_almost_equal < 0.005 * raster.width * raster.height
+
+                # Replace these exceptions by zero in both
+                gdal_proximity[ind_not_almost_equal] = 0.0
+                geoutils_proximity[ind_not_almost_equal] = 0.0
+                # Check that all the rest is almost equal
+                assert np.allclose(gdal_proximity, geoutils_proximity, atol=1e-04, equal_nan=True)
+
+            # Otherwise, results are exactly equal
+            else:
+                assert np.array_equal(gdal_proximity, geoutils_proximity, equal_nan=True)
+
+        # For debugging
+        except Exception as exception:
+
+            import matplotlib.pyplot as plt
+
+            # Plotting the xdem and GDAL attributes for comparison (plotting "diff" can also help debug)
+            plt.subplot(121)
+            plt.imshow(gdal_proximity)
+            # plt.imshow(np.abs(gdal_proximity - geoutils_proximity)>0.1)
+            plt.colorbar()
+            plt.subplot(122)
+            plt.imshow(geoutils_proximity)
+            # plt.imshow(raster.data.data == 112)
+            plt.colorbar()
+            plt.show()
+
+            # ind_not_equal = np.abs(gdal_proximity - geoutils_proximity)>0.1
+            # print(gdal_proximity[ind_not_equal])
+            # print(geoutils_proximity[ind_not_equal])
+
+            raise exception
+
+    def test_proximity_parameters(self) -> None:
+        """
+        Test that new (different to GDAL's) proximity parameters run.
+        No need to test the results specifically, as those rely entirely on the previous test with GDAL,
+        and tests in rasterize and shapely.
+        #TODO: Maybe add one test with an artificial vector to check it works as intended
+        """
+
+        # -- Test 1: with self's Raster alone --
+        raster1 = gu.Raster(self.landsat_b4_path)
+        prox1 = raster1.proximity()
+
+        # The raster should have the same extent, resolution and CRS
+        assert raster1.crs == prox1.crs
+        assert raster1.transform == prox1.transform
+
+        # It should change with target values specified
+        prox2 = raster1.proximity(target_values=[255])
+        assert not np.array_equal(prox1.data, prox2.data)
+
+        # -- Test 2: with a vector provided --
+        vector = gu.Vector(self.everest_outlines_path)
+
+        # With default options (boundary geometry)
+        prox3 = raster1.proximity(vector=vector)  # noqa
```

Comment:
What's with the `# noqa`?

Reply:
Because of `flake8` warning on `prox3`, `prox4`, etc... not being used. I just removed them and simply ran the functions.